docs/_drafts/scv2-operator.md

```diff
@@ -0,0 +1,176 @@
+However, this coupling runs deep, with various aspects of ML deployments understood in the form of Kubernetes resources and with no support for other container orchestrators.
+
+Let's briefly cover how Core v1 manages deployments.
+The sole Kubernetes custom resource is the [SeldonDeployment](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L226), which acts as a wrapper around all the functionality.
+A `SeldonDeployment` comprises one or more [predictors](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L229), such as a main deployment and a shadow or canary.
+Each predictor is an independently-deployed inference graph composed of so-called [predictive units](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L245) arranged in a tree structure.
+Each predictive unit represents some [type of functionality](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L539) and can have zero or more [children](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L601), which are also predictive units.
+Entire predictors are represented by Kubernetes pods, with each predictive unit deployed as its own container within that pod.
+There is typically an additional container in each predictor's pod for an executor component, which acts as an **orchestrator** for how requests traverse that predictor's inference graph.
+Note that components defined outside a predictor's graph, such as explainers, are deployed in separate pods.
+
+In itself, this orchestrated workflow could be deployed in something other than Kubernetes -- the executor just needs to be able to communicate with each of its predictor's containers.
+However, the way this executor is implemented, it demands to be in a Kubernetes environment.
+We can see [here](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L539) that it wants to know the name of its `SeldonDeployment` (or `SDep` for short) and will [fail](https://github.com/SeldonIO/seldon-core/blob/46ef14e60e1f8c5b4c30320b06293e8b60f721fa/operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go#L539) to start up without this or the Kubernetes namespace.
+Even were this not the case, the endpoints the executor uses to call predictive units may be the Kubernetes `Service` names created automatically by the Core v1 operator, for example when the executor (or "engine") is deployed separately from its predictor.
```

Comment:
@ukclivecox I've never quite wrapped my head around the difference between the engine and the executor.  From what I can glean from the code, the _executor_ is the actual application itself, including when deployed as part of a predictor's pod, whereas the  _engine_ refers to a separate deployment of the executor.  Is that accurate, or is there a bit of a hangover of naming sticking around despite things being deprecated?